bar.m

```diff
@@ -38,76 +38,30 @@
-// because if some thread is executing this method, Hammerspoon is already crashing. While swizzling
-// in a method invoked when this module is loaded rather then when Hammerspoon is first loaded
-// is not ideal, I only want this work around installed when it's actually needed, i.e. when the
-// touchbar module is being used. It's a trade-off and I hope I've judged things correctly -- we'll
-// see...
-+ (void)xxx_hammerspoon_SwizzleCrashOnException {
++ (void)xxx_hammerspoon_SwizzleMarkActiveFunctionRowsAsDimmed {
     static dispatch_once_t onceToken;
     dispatch_once(&onceToken, ^{
         Class targetClass = [self class] ;
 
-        SEL        originalSelector       = NSSelectorFromString(@"_crashOnException:") ;
+        SEL        originalSelector       = NSSelectorFromString(@"markActiveFunctionRowsAsDimmed:") ;
         Method     originalMethod         = class_getInstanceMethod(targetClass, originalSelector) ;
         IMP        originalImplementation = method_getImplementation(originalMethod) ;
         const char *originalEncoding      = method_getTypeEncoding(originalMethod) ;
 
-        SEL        swizzledSelector       = @selector(xxx_hammerspoon_crashOnException:) ;
+        SEL        swizzledSelector       = @selector(xxx_hammerspoon_markActiveFunctionRowsAsDimmed:) ;
         Method     swizzledMethod         = class_getInstanceMethod(targetClass, swizzledSelector) ;
         IMP        swizzledImplementation = method_getImplementation(swizzledMethod) ;
         const char *swizzledEncoding      = method_getTypeEncoding(swizzledMethod) ;
 
-        // will likely be false because originalSelector already exists, but just in case
-        // it goes away in a future macOS update (it is undocumented, after all) lets not
-        // crash when method_exchangeImplementations is invoked. True, this will end up
-        // having no effect then, but we're no worse off than before...
-        BOOL didAddMethod = class_addMethod(
-                                               targetClass,
-                                               originalSelector,
-                                               swizzledImplementation,
-                                               swizzledEncoding
-                                           ) ;
-
-        if (didAddMethod) {
-            class_replaceMethod(
-                targetClass,
-                swizzledSelector,
-                originalImplementation,
-                originalEncoding
-            ) ;
-            [LuaSkin logDebug:[NSString stringWithFormat:@"_crashOnException and xxx_hammerspoon_crashOnException added to %s (probably means this fix ineffective)", class_getName(targetClass)]] ;
-        } else {
-            method_exchangeImplementations(originalMethod, swizzledMethod);
-            [LuaSkin logDebug:[NSString stringWithFormat:@"_crashOnException and xxx_hammerspoon_crashOnException swizzled in %s", class_getName(targetClass)]] ;
-        }
```

Comment:
Minus my comments, which tbh are wrong/incomplete, you should really include the whole process -- what if the method you're swizzling is inherited and not implemented directly within the class?

Reply:
To be honest, the only reason I got rid of this code is because it was trigging "probably means this fix ineffective" - but that's probably because of "category, not a class definition" above.

Reply:
Fixed!